src/contexts/cart&wishlist/CartWishlistProvider.jsx

```diff
@@ -0,0 +1,81 @@
+import {
+  useState,
+  createContext,
+  useContext,
+  useEffect,
+  useReducer,
+} from "react";
+import { useAuth } from "contexts";
+import { useAxios } from "utilities";
+import { cartWishlistReducer } from "./reducer";
+
+const cartWishlistContext = createContext({
+  cartItems: [],
+  wishlistItems:[],
+  setCartItems: () => {},
+});
+
+export const useCartWishlist = () => useContext(cartWishlistContext);
+
+export const CartWishlistProvider = ({ children }) => {
+  const [cartItems, setCartItems] = useState([]);
+  const [wishlistItems, setwishlistItems ] = useState([]);
+  const {
+    userState: { isUserAuthenticated, encodedToken },
+  } = useAuth();
+
+  const [cartWishlistState, cartWishlistDispatch] = useReducer(cartWishlistReducer, {
+    apiUrl: "",
+    apiMethod: "get",
+    postData: {
+      product: {},
+    },
+  });
+  const { serverResponse, isLoading } = useAxios(
+    cartWishlistState.apiUrl,
+    cartWishlistState.apiMethod,
+    cartWishlistState.postData,
+    encodedToken
+    );
+  useEffect(() => {
+    console.log(serverResponse)
+    // To update the cart and wishlist state everytime the app gets new data from the server
+    if(serverResponse.data?.cart){
+      const cart = serverResponse.data?.cart || [];
+      setCartItems(cart);
+    } else if (serverResponse.data?.wishlist){
+      const wishlist = serverResponse.data.wishlist;
+      setwishlistItems(wishlist);
+    }
+  }, [serverResponse]);
+
+  useEffect(() => {
+    let timeoutId;
+    if (!isUserAuthenticated) {
+      cartWishlistDispatch({ type: "clearCart" })
+      setCartItems([]);
+      setwishlistItems([]);
+    }
+    else {
+      console.log({isUserAuthenticated})
+      cartWishlistDispatch({ type: "getCartList" });
```

Comment:
here also

Reply:
Thanks for pointing it out, I have removed it in the latest commit